stdlib/public/runtime/Metadata.cpp

```diff
@@ -2749,6 +2749,14 @@ _swift_updateClassMetadataImpl(ClassMetadata *self,
   // If we're running on a older Objective-C runtime, just realize
   // the class.
   if (!requiresUpdate) {
+    // If we don't have a backward deployment layout, we cannot proceed here.
+    if (self->getInstanceSize() == 0 ||
+        self->getInstanceAlignMask() == 0) {
+      fatalError(0, "class %s does not have a fragile layout; "
+                 "the deployment target was newer than this OS\n",
+                 self->getDescription()->Name.get());
```

Comment:
If we really want, there's a hidden dyld function `dyld_get_min_os_version` that can print what deployment target a library was compiled with. But I don't think that's actually going to be the failure mode; more likely it's a compiler bug where we think the class doesn't need a fragile layout, but then it somehow gets into ObjC initialization anyway (maybe through NSClassFromString).

Reply:
Yes, this error indicates something went very wrong.